indexing-service/src/main/java/io/druid/indexing/overlord/TaskQueue.java

```diff
@@ -378,6 +384,7 @@ private void notifyStatus(final Task task, final TaskStatus taskStatus)
       for (int i = tasks.size() - 1; i >= 0; i--) {
         if (tasks.get(i).getId().equals(task.getId())) {
           removed++;
+          taskLockbox.remove(tasks.get(i));
```

Comment:
remove return val never checked. Suggest adding in log (info maybe?) if task is attempted to remove but not removed.